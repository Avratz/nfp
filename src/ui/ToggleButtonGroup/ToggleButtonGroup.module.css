.toggleButtonGroup {
  display: flex;
  gap: 0.75rem;
  width: 100%;
}
