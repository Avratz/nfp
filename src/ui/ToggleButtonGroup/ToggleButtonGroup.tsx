import { RadioGroup } from '@headlessui/react'

import styles from './ToggleButtonGroup.module.css'

type ToggleButtonGroupProps = {
  value: string
  onChange: (value: string) => void
  children: React.ReactNode
}

export function ToggleButtonGroup({
  value,
  onChange,
  children,
}: ToggleButtonGroupProps) {
  return (
    <RadioGroup
      value={value}
      onChange={onChange}
      className={styles.toggleButtonGroup}
    >
      {children}
    </RadioGroup>
  )
}
