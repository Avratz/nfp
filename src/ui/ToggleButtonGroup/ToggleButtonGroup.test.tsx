import { render } from '@testing-library/react'
import { ToggleButtonGroup } from './ToggleButtonGroup'

describe('ToggleButtonGroup', () => {
  it('renders default', () => {
    const { container } = render(
      <ToggleButtonGroup value="ToggleButtonGroup" onChange={() => null}>
        ToggleButtonGroup
      </ToggleButtonGroup>,
    )
    expect(container).toMatchSnapshot()
  })
})
