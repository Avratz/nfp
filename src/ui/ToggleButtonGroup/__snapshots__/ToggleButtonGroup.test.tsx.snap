// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButtonGroup renders default 1`] = `
<div>
  <div
    class="toggleButtonGroup"
    id="headlessui-radiogroup-:r0:"
    role="radiogroup"
  >
    ToggleButtonGroup
  </div>
</div>
`;
